# 使用 Anaconda

Anaconda 是一个很好用的 Python IDE，它集成了很多科学计算需要使用的python第三方工具包。

## conda 的使用

根据自己的操作系统安装好 Anaconda 后，在命令行下输入：

conda list

可以看已经安装好的 python 第三方工具包。

也可以在 ipython cell 中用 !conda list 命令来查看。

In [2]:
!conda list

# packages in environment at /Users/cmh/anaconda3:
#
# Name                    Version                   Build  Channel
_ipyw_jlab_nb_ext_conf    0.1.0            py36h2fc01ae_0  
alabaster                 0.7.10           py36h174008c_0  
anaconda                  custom           py36ha4fed55_0  
anaconda-client           1.6.14                   py36_0  
anaconda-navigator        1.8.3                    py36_0  
anaconda-project          0.8.2            py36h9ee5d53_0  
appnope                   0.1.0            py36hf537a9a_0  
appscript                 1.0.1            py36h9e71e49_1  
asn1crypto                0.24.0                   py36_0  
astroid                   1.6.1                    py36_0  
astropy                   2.0.3            py36h1de35cc_0  
attrs                     17.4.0                   py36_0  
babel                     2.5.3                    py36_0  
backports                 1.0              py36ha3c1827_1  
backports.shutil_get_terminal_size 1.0.0

可以使用它来安装，更新，卸载第三方的 python 工具包：

conda install <some package>
conda update <some package>
conda remove <some package>

在安装或更新时可以指定安装的版本号，例如需要使用 numpy 1.8.1：

conda install numpy=1.8.1
conda update numpy=1.8.1


查看 conda 的信息：

conda info

In [9]:
!conda info


     active environment : None
       user config file : /Users/cmh/.condarc
 populated config files : /Users/cmh/.condarc
          conda version : 4.5.1
    conda-build version : 3.4.1
         python version : 3.6.4.final.0
       base environment : /Users/cmh/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /Users/cmh/anaconda3/pkgs
                          /Users/cmh/.conda/pkgs
       envs directories : /Users/cmh/anaconda3/envs
                

## 自定义 Python 环境
一个很棒的功能是 conda 可以产生一个自定义的环境，假设在安装的是 Python 2.7 的情况下，想使用 Python 3.4，只需要在命令行下使用 conda 产生一个新的环境：

conda create -n py34 python=3.4

这里这个环境被命名为 py34 ，可以根据喜好将 py34 改成其他的名字。

使用这个环境时，只需要命令行下输入：

    activate py34 #(windows)

    source activate py34 #(linux, mac)

此时，我们的 Python 版本便是 python 3.4了。

## spyder 编辑器

Anaconda 默认使用的编辑器是 spyder，可以在命令行下输入：

spyder

来进入这个编辑器，具体使用方法不做介绍。